### ISSUE TITLE
`TextArea` still uses `Escape` to move focus on

### ISSUE THREAD

**Issue by davep** [Textualize/textual#4110]
While "out of the box" `TextArea` now uses <kbd>Tab</kbd> to shift focus, it also still uses <kbd>Escape</kbd>, ideally it would not when in this mode. Pressing <kbd>Escape</kbd> to exit a modal dialog would be common, so having <kbd>Escape</kbd> do something different given one particular widget type would be surprising for the user.

For example, this dialog uses <kbd>Escape</kbd> as the quick exit key, but instead if moves focus if you happen to press it while in the `TextArea`:

![Screenshot 2024-02-02 at 21 18 15](https://github.com/Textualize/textual/assets/28237/7b1e295e-5908-4afe-94e2-847b20e3f851)

**Comment by darrenburns** [comment]
Yeah I realised this on Thursday. Since we don't have dynamic key bindings I think we'll have to just manage the escape press in "on_key" instead of as a binding.

**Comment by willmcgugan** [comment]
You can raise `SkipAction` from an action, which allow bindings from parents to run. If that helps.
